python/cudf/cudf/tests/test_indexing.py

```diff
@@ -286,6 +286,7 @@ def test_dataframe_loc(scalar, step):
             "d": np.random.random(size).astype(np.float64),
         }
     )
+    pdf.index.name = "index"
```

Comment:
`.iloc` behaved correctly before the changes in this PR. Just adding this here for completeness.